tests/unit/amo/components/TestAddonCompatibilityError.js

```diff
@@ -33,72 +29,106 @@ describe('AddonCompatibilityError', () => {
     };
     const { store } = createStore({ api });
 
-    return findRenderedComponentWithType(renderIntoDocument(
-      <Provider store={store}>
-        <I18nProvider i18n={getFakeI18nInst()}>
-          <AddonCompatibilityError minVersion={null} {...props} />
-        </I18nProvider>
-      </Provider>
-    ), AddonCompatibilityError);
+    const defaultProps = {
+      i18n: getFakeI18nInst(),
+      minVersion: null,
+      store,
+    };
+
+    return shallowUntilTarget(
+      <AddonCompatibilityError {...defaultProps} {...props} />,
+      AddonCompatibilityErrorBase
+    );
   }
 
   it('renders a notice for non-Firefox browsers', () => {
     const root = render({
-      lang: 'es',
+      lang: 'en-GB',
       reason: INCOMPATIBLE_NOT_FIREFOX,
-      userAgentInfo: { browser: { name: 'Chrome' }, os: {} } });
-    const rootNode = findDOMNode(root);
+      userAgentInfo: { browser: { name: 'Chrome' }, os: {} },
+    });
+
+    expect(
+      root.find('.AddonCompatibilityError').render().find('a').attr('href')
```

Comment:
That's interesting. I've been doing this more like `expect(root.find('.localized').html()).toContain('<a ...')` but I like your approach better.